src/atoms/TextInput/TextInput.web.js

```diff
@@ -1,3 +1,486 @@
+    }
+  } else {
+    return 'transparent';
+  }
+};
+
+// eslint-disable-next-line complexity
+const TextInput = ({
+  placeholder,
+  onChange,
+  onBlur,
+  helpText,
+  successText,
+  errorText,
+  variant,
+  prefix,
+  suffix,
+  disabled,
+  value,
+  iconLeft,
+  iconRight,
+  maxLength,
+  label,
+  testID,
+  labelPosition,
+  width,
+  type,
+  id,
+  name,
+}) => {
+  const theme = useContext(ThemeContext);
+  const inputRef = useRef();
+  const containerRef = useRef();
+  const [isFocused, setIsFocused] = useState(false);
+  const [input, setInput] = useState(value || '');
+  // Used for storing layout value of TextInput
+  const [layoutDimensions, setLayoutDimensions] = useState(null);
+  // Used to hide placeholder while label is inside the TextInput
+  const [isPlaceholderVisible, setIsPlaceholderVisible] = useState(isFocused);
+
+  const hasText = !!(isDefined(input) && input.length > 0);
+  const hasAnimatedLabel = variant === 'outlined';
+
+  const placeholderTextColor = getPlaceholderTextColor({
+    theme,
+    disabled,
+    isPlaceholderVisible,
+    hasAnimatedLabel,
+  });
+
+  // Derive accessory conditions based on props
+  const { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon } = getAccessoryConfig({
+    errorText,
+    prefix,
+    suffix,
+    iconLeft,
+    iconRight,
+  });
+
+  if (labelPosition === 'left' && variant === 'outlined') {
+    // Outlined Text Input has only a top label
+    throw Error('Cannot have a left label on an outlined Text Input');
+  }
+
+  if (!isEmpty(prefix) && !isEmpty(iconLeft)) {
+    throw Error('Cannot have prefix and left icon together');
+  }
+
+  if (!isEmpty(suffix) && !isEmpty(iconRight)) {
+    throw Error('Cannot have suffix and right icon together');
+  }
+
+  const onFocus = useCallback(() => {
+    setIsFocused(true);
+
+    /* Wait for 90ms to show the placeholder since it takes 100ms for Label to animate from inside to top of the TextInput.
+       Otherwise they both overlap */
+    /* Don't have any delay if label is on left of TextInput */
+
+    setTimeout(() => {
+      setIsPlaceholderVisible(true);
+    }, 90);
+  }, []);
+
+  const onBlurText = useCallback(
+    (event) => {
+      setIsFocused(false);
+      setIsPlaceholderVisible(false);
+      const text = event.target.value;
+      setInput(text);
+      if (onBlur) {
+        onBlur(text);
+      }
+    },
+    [setIsFocused, setIsPlaceholderVisible, setInput, onBlur],
+  );
+
+  const onChangeText = useCallback(
+    (event) => {
+      const text = event.target.value;
+      if (text.length > maxLength) {
+        return;
+      }
+      setInput(text);
+      if (onChange) {
+        onChange(text);
+      }
+    },
+    [maxLength, onChange, setInput],
+  );
+
+  const onSelectText = () => {
+    return theme.colors.primary[980];
+  };
+
+  const onKeyPress = (e) => {
+    if (type === 'number') {
+      const charCode = typeof e.which === 'number' ? e.which : e.keyCode;
```

Comment:
Why is this check needed? Don't we always have `keyCode` available?

Reply:
Yes. `keyCode` returns 0 in firefox.